wifiphisher/common/interfaces.py

```diff
@@ -490,12 +470,16 @@ def set_interface_mac(self, interface_name, mac_address):
         # card must be turned off(down) before setting mac address
         try:
             pyw.macset(card, mac_address)
+            return True
         # make sure to catch an invalid mac address
         except pyric.error as error:
             if error[0] == 22:
-                raise InvalidMacAddressError(mac_address)
+                message = "The provided MAC address {0} is invalid".format(
+                    mac_address)
             else:
-                raise
+                message = "Fail to set mac address {0}".format(mac_address)
+            print('[' + constants.R + '!' + constants.W + ']' + message)
```

Comment:
My only concern is the `print` statement here. Generally, to keep things tidy, we want to `print` only through our main controller which is `pywifiphisher`. I know this is not the case in our codebase but I believe we should stick to this rule from now on.

That means that instead of printing here, I believe it's better if you define a method in `pywifiphisher` `try_change_mac` that will call `set_interface_mac` and in case of an exception, print the message.

Reply:
Good idea. I will change it today.